src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -24,6 +24,8 @@
  */
 class Connection
 {
+    public const ATTEMPT_COUNT_HEADER_NAME = 'symfony-messenger-attempts';
```

Comment:
AMQP uses `x-death` when a message is dead-letter-ed. Maybe we can reuse the same? (not sure if it conflicts though, would need to try).

Reply:
Hmm. I don't know much about this, but I don't think it'll work. Each time we retry, we're re-using the headers from the existing message. This has a nice effect that the `x-death` header (which is an array) will have 0 items, then 1 item, then 2 items, etc - it'll increase with each "death". The "newest" death apparently (i've just tested) always becomes the `0` key - the others are "pushed back". If we try to insert an entry into `x-death`, it just looks like there was a "previous" death, and Rabbit pushes a new item onto the 0 index and our entry is pushed back to 0.

It *seems* like we shouldn't be setting values onto `x-death`.